govcd/vapp_network_test.go

```diff
@@ -0,0 +1,150 @@
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].DestinationIP, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].SourcePortRange, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].SourceIP, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].Protocols.Any, Equals, true)
+
+	check.Assert(result.Configuration.Features.FirewallService.DefaultAction, Equals, "drop")
+	check.Assert(result.Configuration.Features.FirewallService.LogDefaultAction, Equals, true)
+
+	//cleanup
+	task, err := vapp.Delete()
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+}
+
+func (vcd *TestVCD) prepareVappWithVppNetwork(check *C, vappName string) (*VApp, string, *types.NetworkConfigSection, error) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	vapp, err := createVappForTest(vcd, vappName)
+	check.Assert(err, IsNil)
+	check.Assert(vapp, NotNil)
+
+	networkName := "Test_UpdateNetworkFirewallRules"
+	description := "Created in test"
+	const gateway = "192.168.0.1"
+	const netmask = "255.255.255.0"
+	const dns1 = "8.8.8.8"
+	const dns2 = "1.1.1.1"
+	const dnsSuffix = "biz.biz"
+	const startAddress = "192.168.0.10"
+	const endAddress = "192.168.0.20"
+	const dhcpStartAddress = "192.168.0.30"
+	const dhcpEndAddress = "192.168.0.40"
+	const maxLeaseTime = 3500
+	const defaultLeaseTime = 2400
+	var guestVlanAllowed = true
+	var fwEnabled = false
+	var natEnabled = false
+	var retainIpMacEnabled = true
+
+	orgVdcNetwork, err := vcd.vdc.GetOrgVdcNetworkByName(vcd.config.VCD.Network.Net1, false)
+	check.Assert(err, IsNil)
+	check.Assert(orgVdcNetwork, NotNil)
+
+	vappNetworkSettings := &VappNetworkSettings{
+		Name:               networkName,
+		Gateway:            gateway,
+		NetMask:            netmask,
+		DNS1:               dns1,
+		DNS2:               dns2,
+		DNSSuffix:          dnsSuffix,
+		StaticIPRanges:     []*types.IPRange{{StartAddress: startAddress, EndAddress: endAddress}},
+		DhcpSettings:       &DhcpSettings{IsEnabled: true, MaxLeaseTime: maxLeaseTime, DefaultLeaseTime: defaultLeaseTime, IPRange: &types.IPRange{StartAddress: dhcpStartAddress, EndAddress: dhcpEndAddress}},
+		GuestVLANAllowed:   &guestVlanAllowed,
+		Description:        description,
+		FirewallEnabled:    &fwEnabled,
+		NatEnabled:         &natEnabled,
+		RetainIpMacEnabled: &retainIpMacEnabled,
+	}
+
+	vappNetworkConfig, err := vapp.CreateVappNetwork(vappNetworkSettings, orgVdcNetwork.OrgVDCNetwork)
+	check.Assert(err, IsNil)
+	check.Assert(vappNetworkConfig, NotNil)
+	return vapp, networkName, vappNetworkConfig, err
+}
+
+func (vcd *TestVCD) Test_GetVappNetworkByNameOrId(check *C) {
+	vapp, networkName, vappNetworkConfig, err := vcd.prepareVappWithVppNetwork(check, "Test_GetVappNetworkByNameOrId")
```

Comment:
Between `Test_GetVappNetworkByNameOrId` and `Test_UpdateNetworkFirewallRules` we are creating and then destroying the same vApp twice.
Why not combining the operations to save a creation?
1. create vApp
2. test network creation
3. run `Test_GetVappNetworkByNameOrId`
4. test network update
5. cleanup

Reply:
* these tests are from different PRs
* these tests are isolated
* changeing involves make them dependent, even at moment they look like possible to reuse